maintainers/scripts/pluginupdate-py/pluginupdate.py

```diff
@@ -421,28 +453,117 @@ def generate_nix(self, _plugins, _outfile: str):
+                is in the input file.
+
+        Returns:
+            True if we should update plugin and False if not.
+        """
+        return plugin.name.replace(".", "-") in to_update
+
+    def get_update(
+        self,
+        input_file: str,
+        output_file: str,
+        config: FetchConfig,
+        to_update: Optional[List[str]],
+    ):
+        if to_update is None:
+            to_update = []
+
+        current_plugins = self.get_current_plugins(config, self.nixpkgs)
+        current_plugin_specs = self.load_plugin_spec(config, input_file)
+
+        cache: Cache = Cache(
+            [plugin for _description, plugin in current_plugins], self.cache_file
+        )
         _prefetch = functools.partial(prefetch, cache=cache)
 
-        def update() -> dict:
-            plugins = self.load_plugin_spec(config, input_file)
+        plugins_to_update = (
+            current_plugin_specs
+            if len(to_update) == 0
+            else [
+                description
+                for description in current_plugin_specs
+                if self.filter_plugins_to_update(description, to_update)
+            ]
+        )
+
+        def update() -> Redirects:
+            if len(plugins_to_update) == 0:
+                log.error(
+                    "\n\n\n\nIt seems like you provided some arguments to `--update`:\n"
+                    + ", ".join(to_update)
+                    + "\nBut after filtering, the result list of plugins is empty\n"
+                    "\n"
+                    "Are you sure you provided the same URIs as in your input file?\n"
+                    "(" + str(input_file) + ")\n\n"
+                )
+                return {}
 
             try:
                 pool = Pool(processes=config.proc)
-                results = pool.map(_prefetch, plugins)
+                results = pool.map(_prefetch, plugins_to_update)
             finally:
                 cache.store()
 
+            print(f"{len(results)} of {len(current_plugins)} were checked")
+            # Do only partial update of out file
+            if len(results) != len(current_plugins):
+                results = self.merge_results(current_plugins, results)
             plugins, redirects = check_results(results)
 
             plugins = sorted(plugins, key=lambda v: v[1].normalized_name)
-            self.generate_nix(plugins, outfile)
+            self.generate_nix(plugins, output_file)
 
             return redirects
 
         return update
 
+    def merge_results(
+        self,
+        current: list[Tuple[PluginDesc, Plugin]],
+        fetched: List[Tuple[PluginDesc, Union[Exception, Plugin], Optional[Repo]]],
+    ) -> List[Tuple[PluginDesc, Union[Exception, Plugin], Optional[Repo]]]:
+        # transforming this to dict, so lookup is O(1) instead of O(n) (n is len(current))
+        result: Dict[
+            str, Tuple[PluginDesc, Union[Exception, Plugin], Optional[Repo]]
```

Comment:
```suggestion
    ) -> list[tuple[PluginDesc, Union[Exception, Plugin], Repo | None]]:
        # transforming this to dict, so lookup is O(1) instead of O(n) (n is len(current))
        result: dict[
            str, tuple[PluginDesc, Union[Exception, Plugin], Repo | None]
```

Reply:
Done in 3c2cc30, thanks!